connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceAdd.java

```diff
@@ -76,82 +77,91 @@
+            final ServiceBuilder<?> dataSourceServiceBuilder =
+                    Services.addServerExecutorDependency(
+                            serviceTarget.addService(dataSourceServiceName, dataSourceService),
+                            dataSourceService.getExecutorServiceInjector(), false)
+                            .addAliases(dataSourceServiceNameAlias)
+                            .addDependency(ConnectorServices.MANAGEMENT_REPOSITORY_SERVICE, ManagementRepository.class,
+                                    dataSourceService.getManagementRepositoryInjector())
+                            .addDependency(SubjectFactoryService.SERVICE_NAME, SubjectFactory.class,
+                                    dataSourceService.getSubjectFactoryInjector())
+                            .addDependency(ConnectorServices.JDBC_DRIVER_REGISTRY_SERVICE, DriverRegistry.class,
+                                    dataSourceService.getDriverRegistryInjector())
+                            .addDependency(ConnectorServices.IDLE_REMOVER_SERVICE)
+                            .addDependency(ConnectorServices.CONNECTION_VALIDATOR_SERVICE)
+                            .addDependency(ConnectorServices.IRONJACAMAR_MDR, MetadataRepository.class, dataSourceService.getMdrInjector())
+                            .addDependency(ConnectorServices.RA_REPOSITORY_SERVICE, ResourceAdapterRepository.class, dataSourceService.getRaRepositoryInjector())
+                            .addDependency(ConnectorServices.BOOTSTRAP_CONTEXT_SERVICE.append(DEFAULT_NAME))
+                            .addDependency(NamingService.SERVICE_NAME);
+            if (jta) {
+                dataSourceServiceBuilder.addDependency(ConnectorServices.TRANSACTION_INTEGRATION_SERVICE, TransactionIntegration.class, dataSourceService.getTransactionIntegrationInjector())
+                        .addDependency(ConnectorServices.CCM_SERVICE, CachedConnectionManager.class, dataSourceService.getCcmInjector());
+
+            }
+            //Register an empty override model regardless of we're enabled or not - the statistics listener will add the relevant childresources
+            if (registration.isAllowsOverride()) {
+                registration.registerOverrideModel(dsName, DataSourcesSubsystemProviders.OVERRIDE_DS_DESC);
+            }
+            startConfigAndAddDependency(dataSourceServiceBuilder, dataSourceService, dsName, serviceTarget, operation);
+
+            dataSourceServiceBuilder.addDependency(driverServiceName, Driver.class,
                     dataSourceService.getDriverInjector());
 
-        dataSourceServiceBuilder.setInitialMode(ServiceController.Mode.NEVER);
+            dataSourceServiceBuilder.setInitialMode(ServiceController.Mode.NEVER);
 
-        dataSourceServiceBuilder.install();
-        driverDemanderBuilder.install();
+            dataSourceServiceBuilder.install();
+            driverDemanderBuilder.install();
```

Comment:
Re-reading WFLY-4294 again, it seems clear the intent is to not have :write-attribute enable/disable without a reload. In that case I don't see why L81-84 in populateModel() are needed.